corgidrp/fluxcal.py

```diff
@@ -0,0 +1,164 @@
+# This module is written to do an absolute flux calibration observing a standard star having CALSPEC data.
+import glob
+import os
+import numpy as np
+from astropy.io import fits, ascii
+from scipy import integrate
+
+def get_filter_name(dataset):
+    """
+    return the name of the transmission curve csv file of used color filter
+    
+    Args:
+        dataset (corgidrp.Dataset): dataset of the observed calstar
+    
+    Returns:
+        str: filepath of the selected filter curve
+    """
+    datadir = os.path.join(os.path.dirname(__file__), "data", "filter_curves")
+    filters = os.path.join(datadir, "*.csv")
+    filter = dataset[0].ext_hdr['CFAMNAME']
+    filter_names = os.listdir(datadir)
+    f_avail = False
+    for name in filter_names:
+        if filter in name:
+            f_avail = True
+            filter_name= os.path.join(datadir, name)
+            break
+        else:
+            pass
+    if f_avail:
+        return filter_name
+    else:
+        raise Exception("there is no filter available with name {0}".format(filter))
+
+def read_filter_curve(filter_filename):
+    """
+    read the transmission curve csv file of the color filters
+    
+    Args:
+        filter_filename (str): file name of the transmission curve data
+    
+    Returns:
+        lambda_nm (np.array): wavelength in unit Angstroem
+        transmission (np.array): transmission of the filter < 1
+    """
+    tab = ascii.read(filter_filename, format='csv', header_start = 3, data_start = 4)
+    lambda_nm = tab['lambda_nm'].data #unit nm
+    transmission = tab['%T'].data
+    return lambda_nm * 10 , transmission/100.
+
+def read_cal_spec(calspec_filename, filter_wavelength):
+    """
+    read the calspec flux density data interpolated on the wavelength grid of the transmission curve
+    
+    Args:
+        calspec_filename (str): file name of the CALSPEC fits file
```

Comment:
For the current target list, please see Tables 1 and 2 here: https://www.spiedigitallibrary.org/conference-proceedings-of-spie/12188/2628627/Absolute-flux-calibrations-for-the-Nancy-Grace-Roman-Space-Telescope/10.1117/12.2628627.full?tab=ArticleLink

We are working on getting the final target tables approved by JPL for sharing - stay tuned! But in the meantime, you can grab those CALSPEC files for those targets.

Reply:
@rzellem: I implemented the possibility in the pipeline step to either input the name of a calspec star or the file path to a fits file. Note: up to now the names are taken from the list of your poster and put in a dictionary associating the according current calspec fits names, and must match exactly. Can/should be made more flexible in future.

Reply:
let's make it a future work to double check this list once we have that.